fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java

```diff
@@ -41,7 +41,7 @@ public void createCatalog(CreateCatalogStmt stmt) throws DdlException {
         // TODO edit log
     }
 
-    public void dropCatalog(String catalogName) {
+    public synchronized void dropCatalog(String catalogName) {
```

Comment:
I will adjust the granularity of `catalogExists()` lock after the final implementation of relevant interface